backend/src/services/course/course.service.ts

```diff
@@ -68,14 +68,17 @@ class Course {
   }
 
   public async getAllWithCategories(
-    filteringOpts: CourseFilteringDto,
-  ): Promise<CourseGetResponseDto[]> {
-    const { categoryKey, title } = filteringOpts;
+    filteringAndPagination: CourseFilteringDto,
+  ): Promise<EntityPagination<CourseGetResponseDto>> {
+    const { categoryKey, title, page, count } = filteringAndPagination;
     const categoryId = await this.getCategoryIdByKey(categoryKey);
+    const ZERO_INDEXED_PAGE = page - 1;
```

Comment:
Let's make it consistent with other `zeroIndexPage` namings
```suggestion
    const zeroIndexPage = page - 1;
```